includes/classes/ExternalConnections/WordPressExternalConnection.php

```diff
@@ -804,65 +804,48 @@ public function check_connections() {
 				$can_post = array();
 
-				$blacklisted_types = [ 'dt_subscription' ];
-
-				foreach ( $types as $type_key => $type ) {
-
-					if ( in_array( $type_key, $blacklisted_types, true ) ) {
-						continue;
-					}
+				$permission_url = untrailingslashit( $this->base_url ) . '/' . self::$namespace . 'distributor/post-types-permissions';
+
+				if ( function_exists( 'vip_safe_wp_remote_get' ) && \Distributor\Utils\is_vip_com() ) {
+					$permission_response = vip_safe_wp_remote_get(
+						$permission_url,
+						false,
+						3,
+						3,
+						10,
+						$this->auth_handler->format_get_args()
+					);
+				} else {
 
-					$link = $this->parse_type_items_link( $type );
-					if ( empty( $link ) ) {
-						continue;
-					}
+					$permission_response = wp_remote_get(
+						$permission_url,
+						$this->auth_handler->format_get_args(
+							array(
+								'timeout' => self::$timeout,
+							)
+						)
+					);
+				}
+				$permission_body = wp_remote_retrieve_body( $permission_response );
 
-					$route = str_replace( untrailingslashit( $this->base_url ), '', $link );
-
-					if ( ! empty( $routes[ $route ] ) ) {
-						if ( in_array( 'GET', $routes[ $route ]['methods'], true ) ) {
-							if ( function_exists( 'vip_safe_wp_remote_get' ) && \Distributor\Utils\is_vip_com() ) {
-								$type_response = vip_safe_wp_remote_get(
-									$link,
-									false,
-									3,
-									3,
-									10,
-									$this->auth_handler->format_get_args()
-								);
-							} else {
-								$type_response = wp_remote_get( $link, $this->auth_handler->format_get_args( array( 'timeout' => self::$timeout ) ) );
-							}
-
-							if ( ! is_wp_error( $type_response ) ) {
-								$code = (int) wp_remote_retrieve_response_code( $type_response );
-
-								if ( 401 !== $code ) {
-									$can_get[] = $type_key;
-								}
-							}
-						}
-
-						if ( in_array( 'POST', $routes[ $route ]['methods'], true ) ) {
-							$type_response = wp_remote_post(
-								$link,
-								$this->auth_handler->format_post_args(
-									array(
-										'timeout' => self::$timeout,
-										'body'    => array( 'test' => 1 ),
-									)
-								)
-							);
-
-							if ( ! is_wp_error( $type_response ) ) {
-								$code = (int) wp_remote_retrieve_response_code( $type_response );
-
-								if ( 401 !== $code ) {
-									$can_post[] = $type_key;
-								}
-							}
-						}
-					}
+				if ( is_wp_error( $permission_response ) || empty( $permission_body ) ) {
+					$output['errors']['no_permissions'] = 'no_permissions';
+				} else {
+					$permissions = json_decode( $permission_body );
+					$can_get     = array_filter(
+						$permissions->can_get,
+						function ( $key ) {
+							return 'dt_subscription' !== $key;
+						},
+						ARRAY_FILTER_USE_KEY
+					);
+					$can_post    = array_filter(
+						$permissions->can_post,
+						function ( $key ) {
+							return 'dt_subscription' !== $key;
+						},
+						ARRAY_FILTER_USE_KEY
+					);
```

Comment:
This actually makes the code harder to read 😅. Using a private method will help increase the readability here.
```suggestion
$can_post = array_filter(
    $permissions->can_post,
    [$this, 'remove_distributor_internal_post_type']
);
```